docs/usage/requests.rst

```diff
@@ -138,3 +138,23 @@ for ``Body`` , by using :class:`RequestEncodingType.MESSAGEPACK <.enums.RequestE
 .. literalinclude:: /examples/request_data/msgpack_request.py
    :caption: msgpack_request.py
    :language: python
+
+
+Custom Request
+--------------
+
+.. versionadded:: 2.7.0
+
+Litestar supports custom ``request_class``es which can be used to further configure the default :class:`Request`.
+The example below illustrates how to implement custom request class for the whole application.
+
+.. literalinclude:: /examples/request_data/custom_request.py
+    :language: python
```

Comment:
```suggestion
.. dropdown:: Example of a custom request at the application level

    .. literalinclude:: /examples/request_data/custom_request.py
        :language: python
```

Reply:
I cant test this at the moment but hiding this in a dropdown might be good to save some space if not needing this info